bdk_keychain/src/keychain_txout_index.rs

```diff
@@ -248,6 +250,61 @@ impl<K: Clone + Ord + Debug> KeychainTxOutIndex<K> {
         }
     }
 
+    /// Convenience method to call [`store_until_unused_count`] for all keychains.
+    ///
+    /// [`store_until_unused_count`]: Self::store_until_unused_count
+    pub fn store_all_until_unused_count(&mut self, unused_count: u32) -> bool {
+        let keychains = self.keychains.keys().cloned().collect::<Vec<_>>();
+        keychains
+            .into_iter()
+            .map(|keychain| self.store_until_unused_count(&keychain, unused_count))
+            .fold(false, |acc, v| acc || v)
+    }
+
+    /// Derives script pubkeys from the descriptor until the unused script count reaches
+    /// `unused_count`.
+    ///
+    /// Returns whether any new script pubkeys were derived.
+    pub fn store_until_unused_count(&mut self, keychain: &K, unused_count: u32) -> bool {
+        if unused_count < 1 {
+            return false;
+        }
+        let up_to = self
+            .last_active_index(keychain)
+            .map(|i| i.saturating_add(unused_count))
+            .unwrap_or(unused_count.saturating_sub(1));
+        self.store_up_to(keychain, up_to)
+    }
+
+    /// Convenience method to call [`prune_unused`] for all keychains.
+    ///
+    /// `prune_from` contains the lowest allowed index to prune from per keychain (default is 0).
+    ///
+    /// [`prune_unused`]: Self::prune_unused
+    pub fn prune_all_unused(&mut self, prune_from: BTreeMap<K, u32>) -> bool {
+        let keychains = self.keychains.keys().cloned().collect::<Vec<_>>();
+        keychains
+            .into_iter()
+            .map(|keychain| self.prune_unused(&keychain, *prune_from.get(&keychain).unwrap_or(&0)))
```

Comment:
This looks like it will remove everything from a keychain if that's not include in the map and `get` returns `None`??

Instead of `self.keychains` I think this should iterate over the `prune_from` map only..

Reply:
Not "everything", it will remove unused. `prune_from` sets where specifically to prune from, if it is not set for a keychain, it will prune all unused (all in our new terminology, prune unused "padding").

If the user would like to prune for specific keychains, they can just use `prune_unused`.

Reply:
Okay, so the intention here is to prune all unused for keychains if their `prune_from` isn't set.. Ya that makes sense.. Maybe it can be included in the doc to make this intention a bitcoin clear.. I was thinking more like if there are 5 keychains, and I only wanna prune 3 of them, then I would use this.. But that's definitely not the case here..